### PR TITLE
[expo-updates][iOS] Allow native debugging of updates

### PR DESCRIPTION
**Pull request by douglowder**
# Why

Currently, if you have a project with expo-updates installed, you need to make a release build for expo-updates to be enabled (debug builds behave like normal RN project debug builds). This makes it somewhat difficult to test and debug updates when things aren’t working as expected.

To make this easier, we want to add a build setting on both platforms to enable updates in debug builds. Ideally, each platform should have just a single setting that is very easy to switch. When this setting is turned on, in a debug build (a) everything updates-related should be fully enabled as if in a release build (see UpdatesPackage and ExpoUpdatesReactDelegateHandler), and (b) RN should generate a bundle.

# How

For iOS, add the environment variable

```sh
export EX_UPDATES_NATIVE_DEBUG=1
```

and then reinstall pods. The `expo-updates` podspec now detects this environment variable, and makes changes so that the debug code that would normally load from the Metro packager is bypassed, and the app is built with the EXUpdates bundle and other dependencies needed to load updates from EAS.

# Test Plan

Manually tested against EAS.

I will investigate the possibility of building the updates E2E test app this way so that the debug build flows can be tested fully in CI.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-updates/ios/EXUpdates/EXUpdatesUtils.h [hunk 1]
```diff
@@ -18,6 +18,7 @@ NS_ASSUME_NONNULL_BEGIN
 + (NSString *)getRuntimeVersionWithConfig:(EXUpdatesConfig *)config;
 + (NSURL *)urlForBundledAsset:(EXUpdatesAsset *)asset;
 + (NSString *)pathForBundledAsset:(EXUpdatesAsset *)asset;
++ (BOOL)nativeDebugging;
```
Comment (esamelson) [inline packages/expo-updates/ios/EXUpdates/EXUpdatesUtils.h:21]: might be good to add a docblock here briefly explaining why/when this is used.

**Review by esamelson — APPROVED** [review]
Cool, this looks good to me! I think the one remaining step would be to document this somewhere, like maybe the debugging updates guide: https://docs.expo.dev/eas-update/debug-updates/ . (Even though it's iOS only at this point, it's still worth documenting, probably with a note mentioning it will be available in SDK 47+).

File: packages/expo-updates/ios/EXUpdates/EXUpdatesUtils.h (hunk 1 above)
Comment (Kudo) [reply packages/expo-updates/ios/EXUpdates/EXUpdatesUtils.h:21]: second this. somehow the `nativeDebugging` is not very clear for what it is. maybe something like `enableUpdatesInDebugBuild`. do it sounds better?

**Review by Kudo — APPROVED** [review]
leaving a nit comment as Eric. the changes look good to me. thanks for the great improvement for scoping the changes only inside expo-updates 👏

**Review by tsapeta — APPROVED** [review]

File: packages/expo-updates/ios/EXUpdates/EXUpdatesUtils.h (hunk 1 above)
Comment (Kudo) [reply packages/expo-updates/ios/EXUpdates/EXUpdatesUtils.h:21]: i just got confused for `EXAppDefines.APP_DEBUG && !EXUpdatesUtils.nativeDebugging()`. how could it happen for a debug build without native debugging. later i realize it means to enable expo-updates autosetup on debug build.

**Pull request merged**